hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/contracts/operation/HederaCreateOperationExternalizer.java

```diff
@@ -87,7 +85,9 @@ public void externalize(MessageFrame frame, MessageFrame childFrame) {
 
         // if a hollow account exists at the alias address, finalize it to a contract
         if (hollowAccountID != null) {
-            finalizeHollowAccountIntoContract(hollowAccountID, updater);
+            final var contractNonce =
+                    updater.get(childFrame.getContractAddress()).getNonce();
+            finalizeHollowAccountIntoContract(hollowAccountID, updater, contractNonce);
```

Comment:
The method `finalizeHollowAccountIntoContract` takes the `updater` that you're getting the nonce out of.  Consider passing the `childFrame.getContractAddress()` into it and computing the needed nonce there instead.  Seems to maybe belong _there_ not _here_.